cmd/influxd/run/config.go

```diff
@@ -306,3 +307,60 @@ func (c *Config) applyEnvOverrides(prefix string, spec reflect.Value, structKey
 	}
 	return nil
 }
+
+// Diagnostics returns a diagnostics representation of Config.
+func (c *Config) Diagnostics() (*diagnostics.Diagnostics, error) {
+	return diagnostics.RowFromMap(map[string]interface{}{
+		"reporting-disabled": c.ReportingDisabled,
+		"bind-address":       c.BindAddress,
+	}), nil
+}
+
+func (c *Config) diagnosticsClients() map[string]diagnostics.Client {
+	// Config settings that are always present.
+	m := map[string]diagnostics.Client{
+		"config": c,
```

Comment:
`c` implements `diagnostics.Client` directly with its own two, top-level config values `reporting-disabled` and `bind-address`, distinct from the other config sections.  The presence of the top-level / unscoped config settings at all is what makes this a little awkward.

Each key-value pair in the map is registered as a diagnostics client. This results in output like:

```
> show diagnostics
name: build
Branch  Build Time Commit  Version
------  ---------- ------  -------
unknown            unknown unknown


name: config
bind-address reporting-disabled
------------ ------------------
:8088        false


name: config-coordinator
log-queries-after max-concurrent-queries max-select-buckets max-select-point max-select-series query-timeout write-timeout
----------------- ---------------------- ------------------ ---------------- ----------------- ------------- -------------
0s                0                      0                  0                0                 0s            10s


name: config-cqs
enabled run-interval
------- ------------
true    1s

...
```